tests/test_oic_message.py

```diff
@@ -529,17 +548,297 @@ def test_id_token():
+    kj.add_symmetric("", 'dYMmrcQksKaPkhdgRNYk3zzh5l7ewdDJ', ['sig'])
+    kj.add_symmetric("https://sso.qa.7pass.ctf.prosiebensat1.com",
+                     'dYMmrcQksKaPkhdgRNYk3zzh5l7ewdDJ', ['sig'])
+    packer = JWT(kj, sign_alg='HS256',
+                 iss="https://sso.qa.7pass.ctf.prosiebensat1.com",
+                 lifetime=3600)
+    _jws = packer.pack(**idt.to_dict())
+    msg = AuthorizationResponse(code="AccessCode289", id_token=_jws)
+    with pytest.raises(CHashError):
+        verify_id_token(msg, check_hash=True, keyjar=kj,
+                        iss="https://sso.qa.7pass.ctf.prosiebensat1.com",
+                        client_id="554295ce3770612820620000")
+
+
+def test_verify_id_token_at_hash():
+    token = 'AccessTokenWhichCouldBeASignedJWT'
+    lhsh = left_hash(token)
+
+    idt = IdToken(**{
+        "sub": "553df2bcf909104751cfd8b2",
+        "aud": [
+            "5542958437706128204e0000",
+            "554295ce3770612820620000"
+            ],
+        "auth_time": 1441364872,
+        "azp": "554295ce3770612820620000",
+        "at_hash": lhsh
+        })
+
+    kj = KeyJar()
+    kj.add_symmetric("", 'dYMmrcQksKaPkhdgRNYk3zzh5l7ewdDJ', ['sig'])
+    kj.add_symmetric("https://sso.qa.7pass.ctf.prosiebensat1.com",
+                     'dYMmrcQksKaPkhdgRNYk3zzh5l7ewdDJ', ['sig'])
+    packer = JWT(kj, sign_alg='HS256',
+                 iss="https://sso.qa.7pass.ctf.prosiebensat1.com",
+                 lifetime=3600)
+    _jws = packer.pack(**idt.to_dict())
+    msg = AuthorizationResponse(access_token=token, id_token=_jws)
+    verify_id_token(msg, check_hash=True, keyjar=kj,
+                    iss="https://sso.qa.7pass.ctf.prosiebensat1.com",
+                    client_id="554295ce3770612820620000")
+
+
+def test_verify_id_token_at_hash_fail():
+    token = 'AccessTokenWhichCouldBeASignedJWT'
+    token2 = 'ACompletelyOtherAccessToken'
+    lhsh = left_hash(token)
+
+    idt = IdToken(**{
+        "sub": "553df2bcf909104751cfd8b2",
+        "aud": [
+            "5542958437706128204e0000",
+            "554295ce3770612820620000"
+            ],
+        "auth_time": 1441364872,
+        "azp": "554295ce3770612820620000",
+        "at_hash": lhsh
+        })
+
+    kj = KeyJar()
+    kj.add_symmetric("", 'dYMmrcQksKaPkhdgRNYk3zzh5l7ewdDJ', ['sig'])
+    kj.add_symmetric("https://sso.qa.7pass.ctf.prosiebensat1.com",
+                     'dYMmrcQksKaPkhdgRNYk3zzh5l7ewdDJ', ['sig'])
+    packer = JWT(kj, sign_alg='HS256',
+                 iss="https://sso.qa.7pass.ctf.prosiebensat1.com",
+                 lifetime=3600)
+    _jws = packer.pack(**idt.to_dict())
+    msg = AuthorizationResponse(access_token=token2, id_token=_jws)
+    with pytest.raises(AtHashError):
+        verify_id_token(msg, check_hash=True, keyjar=kj,
+                        iss="https://sso.qa.7pass.ctf.prosiebensat1.com",
+                        client_id="554295ce3770612820620000")
+
+
+def test_verify_id_token_missing_at_hash():
```

Comment:
There should be a test for missing `c_hash` as well.

Reply:
Added one